### PR TITLE
Support `defaultItem` as a function inside `add-block-menu.tsx`

### LINKED ISSUE #1640: Function supplied to `defaultItem` is not executed (using inline-blocks-demo)
## Description

[The docs](https://tina.io/docs/ui/inline-editing/inline-blocks/#part-2-block-template) say that the `defaultItem` key on the `BlockTemplate` interface can be either an object or a function that returns an object. 

image

However, when trying to supply a function to `defaultItem`, the function is not executed and the created block is not populated with any data.

### Steps to reproduce

1. Clone and checkout the `final-state` branch of the `inline-blocks-demo`
```bash
git clone https://github.com/tinacms/inline-blocks-demo.git
cd inline-blocks-demo
git checkout final-state
yarn install
yarn dev
```

2. Open `inline-blocks-demo\src\components\FeatureList.js` and modify `defaultItem` to be a function
```javascript
function FeatureList({ index }) { ... }

const FEATURE_BLOCKS = { ... };

export const featureListBlock = {
  ...
  template: {
    ...
    defaultItem: () => {
      console.log("This should print to console")
      return { ... }
    },
    fields: [],
  },
};
```

3. Save the file
4. Add a new feature list block at the bottom of the page

### Expected result

- The feature list is created and populated with default values 
- "This should print to console" appears in developer console

### Actual result

- The feature list is created but is not populated with any values
- Nothing is printed to developer console

#### Tinacms Package Versions
See package.json of https://github.com/tinacms/inline-blocks-demo

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser |   Chrome      |
| os      |  Windows       |
| node    |   12.18.2      |
| npm     |   6.14.5      |

### PR DESCRIPTION
**Pull request by sneyed**
Fixes #1640

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ npm run build
 
 # Start Next.js Demo
 cd packages/demo-next
-npm run dev
+npm run dev-next
```
Comment (spbyrne) [inline CONTRIBUTING.md:24]: Hey, was this change intentional?

File: packages/react-tinacms-inline/src/blocks/add-block-menu.tsx [hunk 2]
```diff
@@ -43,6 +43,10 @@ export function AddBlockMenu({
   const addBlockButtonRef = React.useRef<HTMLButtonElement>(null)
   const [openTop, setOpenTop] = React.useState(false)
 
+  const getDefaultProps = (defaultItem: any) => {
+    return typeof defaultItem === 'function' ? defaultItem() : defaultItem
+  }
```
Comment (spbyrne) [inline packages/react-tinacms-inline/src/blocks/add-block-menu.tsx:48]: Looks good! This is exactly what we needed.

**Review by spbyrne — CHANGES_REQUESTED** [review]
The defaultItem  specific code looks great, just wondering about the `CONTRIBUTING.md` changes.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kendallstrautman) [reply CONTRIBUTING.md:24]: Good catch, it could be this or `develop` I forgot we removed the `dev` script

Comment (sneyed) [reply CONTRIBUTING.md:24]: Yes, it was intentional, although not strictly related to fixing #1640 😄 
Just noticed the run command was not correct while working on this change

Comment (sneyed) [reply CONTRIBUTING.md]: Reverted 910f1411632773b70f2078da006e7ad8f2888ff4 to exclude this change

**Review by spbyrne — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (spbyrne) [reply CONTRIBUTING.md]: Ah, my bad. Thanks!
